File: VB/FolderBrowserDialogServiceSample/FolderBrowserDialogViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.DataAnnotations

Namespace FolderBrowserDialogServiceSample.ViewModel

    Public Class FolderBrowserDialogViewModel
        Inherits ViewModelBase

        Public Property Description As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property ShowNewFolderButton As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property

        Public Property ResultPath As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public ReadOnly Property FolderBrowserDialogService As IFolderBrowserDialogService
            Get
                Return GetService(Of IFolderBrowserDialogService)()
            End Get
        End Property

        Public Sub New()
            ShowNewFolderButton = True
        End Sub

        <Command>
        Public Sub ShowDialog()
            If FolderBrowserDialogService.ShowDialog() Then ResultPath = FolderBrowserDialogService.ResultPath
        End Sub
    End Class
End Namespace

File: CS/FolderBrowserDialogServiceSample/FolderBrowserDialogViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.DataAnnotations;

namespace FolderBrowserDialogServiceSample.ViewModel {
    public class FolderBrowserDialogViewModel : ViewModelBase {
        public string Description { get { return GetValue<string>(); } set { SetValue(value); } }
        public bool ShowNewFolderButton { get { return GetValue<bool>(); } set { SetValue(value); } }
        public string ResultPath { get { return GetValue<string>(); } set { SetValue(value); } }
        public IFolderBrowserDialogService FolderBrowserDialogService { get { return GetService<IFolderBrowserDialogService>(); } }

        public FolderBrowserDialogViewModel() {
            ShowNewFolderButton = true;
        }
        [Command]
        public void ShowDialog() {
            if (FolderBrowserDialogService.ShowDialog())
                ResultPath = FolderBrowserDialogService.ResultPath;
        }
    }
}
